src/main/java/ResourceDiscovery/AssetObjects/AssetObject.java

```diff
@@ -182,32 +203,61 @@ public String getStatus() {
         return this.status;
     }
 
+    /**
+     * Get the AssetType of this object.
+     * @return A enum of AssetTypes representing the specific type of this Asset Object.
+     */
+    public AssetTypes getAssetTypeEnum() {
+        return this.assetTypeEnum;
+    }
+
     /*
      * This function receives a string representing a url and extracts the string after the last '/'
      * char in the url.
      * If the provided string does not match that pattern the original string is returned.
      */
-    protected static String getLastSeg(String url) {
-        Matcher matcher = LAST_SEGMENT_PATTERN.matcher(url);
+    protected static String getLastSeg(Object url) {
+        String urlToReturn = (String) url;
+        Matcher matcher = LAST_SEGMENT_PATTERN.matcher(urlToReturn);
```

Comment:
IIUC 'matcher' should be "greedy" in order for this to work as decribed?
Are you sure it is?
Can you rephrase this in a way that will remove this ambiguity?

Reply:
also I think it would be more readable if it didn't require multiple matches.